### ISSUE TITLE
Surveys: Coordination

### ISSUE THREAD

**Issue by SMotaal** [nodejs/modules#175]
📆
Last Meeting
<sup>Thursday September 19 8-9 PM (UTC)</sup>

Thanks everyone for joining… I will try to post some updates for those who missed before our next regular meeting time.

---

**Tasks**

- [x] 🎉 **A1 (closed)** @SMotaal (@inidaname @sendilkumarn) — Sep 3
- [x] ✋ **A2 (closed)** @sendilkumarn (@GeoffreyBooth @zenparsing)  — Sep 10
- [x] 🎉 **B1 (internal)** @zenparsing @sendilkumarn @devsnek — Sep 12
- [ ] 🔨 **Cx (collaborators)** — due Oct 5
- [ ] 🔨 **Dx (developers)** 
   - [ ] 🔨Activities & Teams
   - [ ] 🔨Purpose & Scope
   - [x] ✋Project Board

<details><summary>More Details<pre> 🎉done   🔨ongoing   ✋pending   ❓reconsidering </pre></summary>

- 🎉 **A1** (participants: @inidaname @sendilkumarn, moderator: @SMotaal)
   - [x] A1 survey
   - [x] A1 review — Fri Aug 31 - Mon Sep 3 
      - [x] A1 (hangouts) Fri 7:00 PM GMT (@SMotaal @inidaname @sendilkumarn)
   - [x] Pick the A2 moderator (@sendilkumarn)
   - [x] Finalize form A2 (@sendilkumarn @SMotaal)
- ✋ **A2** (participants: @GeoffreyBooth @zenparsing, moderator: @sendilkumarn)
   - [x] Send invites to A2 (@sendilkumarn)
   - [x] 🎉 A2 survey (also open to A1 participants) — Mon Sep 3 - Sep 5
   - [x] 🔨 A2 review (also open  to A1 participants) — Wed Sep 5 - Sep 7
      - [x] A2 (doodle)
      - [x] create forked A3 draft for review (@SMotaal)
      - [x] 🎉 A2 hangouts 1/2 TBD (@sendilkumarn @zenparsing @SMotaal)
      - ❓ A2 hangouts 2/2 TBD (@sendilkumarn @GeoffreyBooth @SMotaal)
   - ❓ Pick the A3 moderator (TBD)
   - ❓ Finalize form A3 (@sendilkumarn @GeoffreyBooth @zenparsing @SMotaal)
- 🎉 **B1** internal survey (moderators: @zenparsing @sendilkumarn @devsnek) — Sep 12
   - [x] 🎉 B1 draft (@zenparsing) — Mon Sep 10
   - [x] 🎉 B1 discussion (@zenparsing @SMotaal) — Tue Sep 11
   - [x] 🎉 B1 generate email list (@sendilkumarn @devsnek)
   - [x] 🎉 B1 send participation links (@sendilkumarn @devsnek) — Tue Sep 11
   - ❓B1 send results links (@SMotaal) — Fri Sep 14

</details>

---
Here is an open link to the most recent "ongoing" draft: [C1: ~~Developers~~ Collaborators Survey](https://goo.gl/wcSZNR)

---

**Important**: Please use direct mentions only

**Comment by SMotaal** [comment]
@inidaname @sendilkumarn 

Let's confirm that we are all available for hangouts today starting 7PM GMT (we'll see how long it takes the first time).

We will be covering proposed changes for A2, how to go about dividing the workload, and what we can do to help mitigate confusion and concerns as we move forward.

**Comment by sendilkumarn** [comment]
The hangout link is available [here](https://hangouts.google.com/call/7NmMJ-d-kaybOLbazO5xAAEI) for today's meeting 

Come and join everyone 👍

**Comment by SMotaal** [comment]
@sendilkumarn I think the whole process works only if people taking the survey do so for the first time, can we please keep the hangout session closed (at least the portion of it that deals with reviewing the questions from A1 -> A2)

**Comment by sendilkumarn** [comment]
@SMotaal Deleted my comment here. Agree to that 👍 

Since there were many people expressing interest in the doodle the other day I thought they might want to join 😉

**Comment by SMotaal** [comment]
@sendilkumarn This is also new to me... I am just trying hard to follow the better practices that I learned regarding survey validity so that our survey succeeds and not become a controversial matter that can potentially lead to very negative outcomes to our collective efforts.

**Comment by sendilkumarn** [comment]
@SMotaal No worries, it is also new to me 👍 
I have sent you both a private link to the email address 👍

**Comment by inidaname** [comment]
Hello didn't get the invite link for hangout

**Comment by SMotaal** [comment]
@inidaname I was happy you were able to join. We're all over as we find a good way to structure this, your efforts are instrumental and highly appreciated.

**Comment by SMotaal** [comment]
@sendilkumarn If you are happy with the A2 form we looked at together (give it a spin yourself following the steps below first) then we can finalize A1 and begin to A2 (and pass moderation from me to you)

**Steps**

Before sending invites to participants make sure you know exactly how many (if any) responses were submitted by the moderator (or reviewers) and make sure anyone with access is made aware that the review stage ended (no more trials). We can use this thread with direct mentions to anyone involved in the review (this time you notify Hassan and myself).

1. Send invites

<img width="591" alt="screen shot 2018-09-02 at 5 20 08 pm" src="https://user-images.githubusercontent.com/849613/44960836-7e4bd980-aed4-11e8-9d4f-571d8d4e972f.png">

2. Monitor the "Responses" tab

<img width="770" alt="screen shot 2018-09-02 at 5 32 27 pm" src="https://user-images.githubusercontent.com/849613/44960918-2f9f3f00-aed6-11e8-95a4-94d022aa3736.png">

3. Once you have the expected number of new responses (ie two) toggle "Accepting responses" off

4. This concludes the survey part

Let's coordinate for the review part once the survey is in motion.

**Comment by SMotaal** [comment]
@sendilkumarn I did a dry run and I have a couple of changes that I want to make to the Internal feedback before we finalize, and a couple of changes to questions which can wait till A2 review.

**Comment by SMotaal** [comment]
@sendilkumarn I noticed that when I'm leave the form window open in Chrome (with some experimental flags on) after leaving the window in the background, somehow the form's model goes out of sync and you can suddenly see the text of questions and headings switched around. **Don't freak out, don't try to correct, just reload the window** because changes in this state simply means you are likely to make irreversible changes that are applied to a different model than the one you are actually seeing. (Very sad design)

**Comment by sendilkumarn** [comment]
@SMotaal Thanks for the tip 👍

**Comment by sendilkumarn** [comment]
@GeoffreyBooth Thanks for filling the survey.
@zenparsing Did you receive the link for the survey, let me know if you have any difficulties.

Also, can we schedule a meeting on  7 or 8 Sep?

**Comment by zenparsing** [comment]
@sendilkumarn Yes, I completed the survey a few hours ago.

**Comment by GeoffreyBooth** [comment]
Sure, I’m available until 8 am Pacific Time any weekday.

**Comment by sendilkumarn** [comment]
@GeoffreyBooth  awesome I just created this [doodle](https://doodle.com/poll/bqgpytcvsk6x5s5q)

@zenparsing can you also give your preference?

**Comment by sendilkumarn** [comment]
Sure, unfortunately, I am not free tomorrow (6th Sep). I am okay with anytime today and Friday.

**Comment by SMotaal** [comment]
@sendilkumarn if you prefer, I am open to doing the review on the 6th and we both can touch base on the 7th for finalization and next steps.

**Comment by sendilkumarn** [comment]
@SMotaal I am perfectly fine with that 👍

**Comment by SMotaal** [comment]
@GeoffreyBooth @zenparsing please take a moment to "re"[doodle](https://doodle.com/poll/bqgpytcvsk6x5s5q) times for tomorrow.

Thanks to everyone's hard work I am feeling very motivated by how this is coming along.

**Comment by SMotaal** [comment]
@sendilkumarn can u doodle yourself too (obviously Friday onwards)

**Comment by SMotaal** [comment]
@GeoffreyBooth There is the one slot on Fri Sept 7th 10-11 Pacific (it's 1-2 Eastern if my 4am Math works)... another Friday 2-3 PDT (5-6 EDT). If those are an option for you, I see now that this may be better than trying to rush it (sorry about the back and forth).

**Comment by SMotaal** [comment]
@GeoffreyBooth @zenparsing With the assumption you both are Pacific Time [I'm Eastern, 3 hrs ahead](https://www.google.ca/search?ei=EfOQW9yoO8Hl_Qb41oHwAw&q=difference+between+pacific+time+and+eastern&oq=difference+between+pacific+time+and+eastern&gs_l=psy-ab.3...13199.16613.0.16830.0.0.0.0.0.0.0.0..0.0....0...1c.1j2.64.psy-ab..0.0.0....0.hORnqt8nboc)

Please clarify:

Thursday & Friday: if you are free before 8 Pacific (11 Eastern)
Saturday & Sunday: if at all

And update doodle accordingly.

@zenparsing I see "?" in doodle for today, does this mean "if need be" or "did not update"

**Comment by zenparsing** [comment]
@SMotaal I've updated my times.

**Comment by SMotaal** [comment]
@sendilkumarn Thursday was not an option... I leave it up to you to pick the ideal Friday time (assuming we either one of us or both can follow up with Geoffrey over the weekend)

My hope is that we at least have enough input and discussion to move on either the internal survey or the draft, even if we get additional input over the weekend.

Does this sound like a plan? When do you want to meet tomorrow (based on doodle)?

**Comment by SMotaal** [comment]
@sendilkumarn @zenparsing 

For do today's review: 👍 to do this ASAP (first slot from doodle ~ 30 mins away)

**Comment by sendilkumarn** [comment]
Lets do it in 2 hours exact. I will send out hangout invitation in an hour 👍

**Comment by sendilkumarn** [comment]
@SMotaal @zenparsing can we meet at Hangout ? here is the link https://hangouts.google.com/call/AMaNEIjurEJzCHabPRePAAEI

**Comment by SMotaal** [comment]
@sendilkumarn @zenparsing today's review was amazing... hope you are also happy with the progress we are all bringing to this effort. Also, I try to keep this issue as up-to-date as I can, if you see something missing or outdated please mention+comment and I will make the edit (not sure if you can make the edits yourself, but if so feel free to make corrections as needed)

**Comment by Fishrock123** [comment]
I hate the way these questions are worded. I refuse to fill it out.

Please adjust it to not have such confrontationally worded answers to questions.

**Comment by zenparsing** [comment]
@Fishrock123 

If you are referring to the internal survey then I humbly apologize for the wording as I am most responsible. I tried to inject a bit of humor (I think we all take ourselves a bit too seriously), but I can see how it can be easily misunderstood.

**Comment by SMotaal** [comment]
Thank you so much @zenparsing for taking the initiative.

I need to remind everyone that people who are taking initiative are doing so for our collective responsibility to roll out a survey. Refusing the contributions of those involved in the this effort is not something this team at large can entertain at the moment. The community at large is eagerly looking at our team to land ES modules. Those modules need to represent the communities expectations.

Please consider collaborating with us to improve the odds of our efforts. The process is ongoing, it is not an individual responsibility and more importantly, accountability is mutual to everyone in the team, the "Modules" team.

**Comment by SMotaal** [comment]
@zenparsing @mcollina @SMotaal @mylesborins @benjamn @mduleone @devsnek @weswigham @targos @ljharb

Thank you so much for taking the first B1 (internal track) survey. Please forgive my inability to address your all concerns during the meeting and in the few emails sent regarding the survey. I need a couple of days to shift gears and come back to this with fresh eyes. But before I do, here are some of the concerns that I believe are important to address before next meeting.

- scope and purpose of internal/developers tracks
- process (and deadlines)
- survey merit

Let me hash out a clear path forward and reach out to each one with direct mentions regarding next steps.

**Comment by SMotaal** [comment]
My take on a process that minimizes the potential for noise and indecision:
- [ ] Topic experts (minimum of 5)
  - [ ] Use responses to decide one "point person" for each technical aspect
  - [ ] Use responses to select one "representer" for each third-party aspect/technology
- [ ] Facilitators (minimum of 3)
  - [ ] Elect ~3 WG members to coordinate our efforts
  - [ ] Confirm with current survey volunteers if they are still on board
- [ ] Developers Track
  - [ ] Define the scope and purpose of Developers Track
  - [ ] Separate draft C1 questions by topic (different forms)
  - [ ] Add respective topic experts as **collaborators** to each *form*
    - [ ] Select 1 **representative** to cross-coordinate with other *forms* 
  - [ ] Wait for **collaborators** from each *form* to:
    - [ ] Conduct revisions (rewrite really) of form (outside regular meeting time please)
    - [ ] Expect **representative** to present revised section to **facilitators**
  - [ ] Wait for **facilitators** to:
    - [ ] Update a single "compiled" by coordinating with **representatives**
    - [ ] Figure collect feedback from others and share it with **representatives**
- [ ] Internal Track
  - [ ] Define the scope and purpose of Internal Track
  - [ ] Consider "halting" till after Vancouver events

**Comment by SMotaal** [comment]
@sendilkumarn @inidaname @GeoffreyBooth Your initiative was vital to getting this off the ground. Please confirm is you are still interested in participation and/or contribution in survey efforts, moving forward.

**Comment by sendilkumarn** [comment]
@SMotaal Let us do it 👍

**Comment by SMotaal** [comment]
@sendilkumarn @GeoffreyBooth @inidaname I realize you filled out (probably multiple) Ax surveys, can you just fill out the very very short form I sent, just because we need to know your aspects interested in working on moving forward

**Comment by SMotaal** [comment]
@GeoffreyBooth are you willing to open and moderate an issue to allow people to openly discuss the "Developers Survey: Scope and Purpose".

We don't want to many issues, but I think this thread is an exception. Everyone wants to be heard, and wants to know what the scope and purpose is.

Our efforts so far were inclined heavily towards deliver a process assuming there was a clear scope and purpose (my bad). As far as we (the survey folks) are relatively far more sensitive to the expectations and perspective of developers, ie users of Node.js, it is really up to the more seasoned Node.js and tooling folks to figure out the scope and purpose that best serves them - each one independently, no conflicts. We get the chore of finding the widest (and most reasonable) scope and purpose based on everyone's inputs.

I apologize, the more I try to be to the point, the more messy I get (that's just how I communicate, generally) so please feel free to back and forth here if I failed to clarify the intent of the thread, but obviously your challenge is finding a more articulate way to express those goals.

If you can't handle this in a day or so, let me know (or recommend someone else to take it over please).

Cheers

**Comment by GeoffreyBooth** [comment]
@SMotaal I’m happy to moderate an issue. I know part of the point is that it’s vague right now, but can you help me define what you mean by “scope and purpose”? Like the scope of the modules implementation that we’re trying to build? And what purpose would it have, other than to bring support for ES modules to Node?

**Comment by SMotaal** [comment]
@GeoffreyBooth 

Let's start by calling this Purpose & Scope:

Purpose of the Developers Survey from the WG's perspective:

I don't think I want to color this one, but drawing upon input from the end user is almost always a way to overcome indecision and barriers to compromise in order to reach some mutually acceptable outcomes.

I think I can try to give you my own formula for the purpose just as an example of what we expect others to express.

> See also https://github.com/nodejs/modules#purpose

It is great that you pointed my attention to the README, because I bookmarked the issues tab.

Since I have no commitments to end users, I look at those features purely from the sense of quality (namely reliability), performance, and ease of use (slash conformance obviously).

In a sense, the developers survey in my world is one that would allow me to extrapolate concrete indicators for the three sides of this triangle regarding specific features. I would expect if to test the end user's position on specific features that would directly affect those three aspects. 

> **Purpose**
> 
> To generate data to support or reject one or more of those features, so that we can overcome indecision by selecting the best set of interdependent features regardless of existing barriers.
> 
> It is also not the purpose to suggest that all the features indicated by the survey should and will be possible, nor that features not indicated will be eliminated (though maybe configurable for end user's benefit).

But that **does not** satisfy the concerns of those who have specific commitments. So in a sense, they can start by saying (which may not be something to be disclose) my commitments are X, Y, Z, and as such, I think that the purpose of this survey is to … but not ….

Scope of the Developers Survey from each party's perspective, upon the completion of the survey:

- we should be able to know … (this is a given)
  - my own formula: Feature(s)/Indicator(s) …
- we don't expect to know … (sometimes important to avoid creep)
  - my own formula: Feature(s)/Indicator(s) …

I think once we can a good enough starting point, this becomes a PR doc for fine tuning.

What are your thoughts?

**Comment by SMotaal** [comment]
@sendilkumarn I think our next steps in general need to grow beyond our direct mentions method and make better use GitHub's tools. This I believe calls for a "ongoing" project(s) board. I have some of the structure in mind, but for now, I would like to ask you to continue your in your amazing self-directed getting-things-done streak to start setting us up.

Can you maybe open a separate issue to:
  - maybe call it "Surveys: a dedicated projects board (for Modules efforts)"
  - invite ideas from everyone on structure… etc
  - ensure people are on board (mention MylesBorins in it please)
  - avoid discussion about outside that issue (let's keep #175 more like HQ)

We need to make sure that this board is the only source of truth moving forward for all activities related to surveys. So, this requires the conventional columns as well as focus columns based on a *concise* logical breakdown of our activities (see below), in my opinion.

<details><summary>Activities (<a href="https://goo.gl/cDM1Uj">Node.js Modules Developers Surveys</a>)</summary>

- General
  - Purpose & Scope
  - Distribution
- Content
  - Flow & Presentation
  - Wording
  - Terminology
  - Representation
- Technical
  - Front-end
  - Back-end
  - Node‍.js
  - Modules
  - Workflow
</details>

**Comment by SMotaal** [comment]
@zenparsing @GeoffreyBooth @mcollina @SMotaal @MylesBorins @benjamn @mduleone @devsnek @weswigham @targos @ljharb @inidaname @Fishrock123

You are mentioned because you indicated interest (or had questions) and because you might have missed our initial steps to arrange a meeting (#183).

Please join us today (#183) where we will have a discussion to try to address all our mutual dilemmas on getting input from developers (the right way).

We will just use a free zoom meeting. I'm trying to see if we can go over the 40 min limit for free.

**Comment by inidaname** [comment]
@SMotaal Hello I have been off this for sometime due to some official assignments, soon rounding up. I will be glad to pick up once I am fully back thank you.
